cmd/azureBlobUpload.go

```diff
@@ -0,0 +1,129 @@
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+)
+
+// AzureContainerAPI is used to mock Azure containerClients in unit tests
+type AzureContainerAPI interface {
+	NewBlockBlobClient(blobName string) (*azblob.BlockBlobClient, error)
+}
+
+// NewBlockBlobClient creates a blockBlobClient from a containerClient
+func NewBlockBlobClient(blobName string, api AzureContainerAPI) (*azblob.BlockBlobClient, error) {
+	return api.NewBlockBlobClient(blobName)
+}
+
+// UploadFile uploads a file to an Azure Blob Storage
+// The function uses the UploadFile function from the Azure SDK
+// We introduce this 'wrapper' for mocking reasons
+func UploadFile(ctx context.Context, api *azblob.BlockBlobClient, file *os.File, o azblob.UploadOption) (*http.Response, error) {
+	return api.UploadFile(ctx, file, o)
+}
+
+// Struct to store Azure credentials from specified JSON string
+type azureCredentials struct {
+	SASToken    string `json:"sas_token"`
+	AccountName string `json:"account_name"`
+	Container   string `json:"container_name"`
```

Comment:
These are all required, right? We could validate them with "github.com/go-playground/validator/v10", as can be seen [in this struct](https://github.com/SAP/jenkins-library/blob/20383907a383efb1c0abe41179d4e0386d00fdce/pkg/ans/event.go#L35-L45). [With the validator being used here](https://github.com/SAP/jenkins-library/blob/20383907a383efb1c0abe41179d4e0386d00fdce/pkg/ans/event.go#L63).

Reply:
Done. I did the check via the "required" tag.